### ISSUE TITLE
Bug / incompatibility with zsh-autosuggestions

### ISSUE THREAD

**Issue by samhh** [subnixr/minimal#5]
Here's what's happening: https://asciinema.org/a/621o9knq5n7gi0nydis4spipi

On that second echo, upon hitting enter 'ing' should disappear. As far as I can tell this isn't a regression but just an uncaught bug/issue.

Here's what should happen, and what does happen if I disable minimal: https://asciinema.org/a/5fiupt2qxl55gtwgpq44k341g

**Comment by subnixr** [comment]
This is very interesting.

I see that the bug disappears when magic enter is disabled https://asciinema.org/a/66q7pcscf27gqmu6103qap2wl

Also, disabling zsh-syntaxhighlighting while enabling magic enter shows a different behavior https://asciinema.org/a/3tha81adfp8inueear0mym2r1

Surely it has something to do with magic enter.

Unfortunately at the moment I have very little time to dedicate, if you find anything, don't esitate to update this issue ;)

**Comment by subnixr** [comment]
Alright, got a couple of hours to spare and found out what's going on.

tl;dr:  `ZSH_AUTOSUGGEST_CLEAR_WIDGETS+=minimal-magic-enter` after sourcing both `minimal` and `zsh-autosuggestions`.

So, I digged around zsh-autosuggestions and zsh documentations about widgets. zsh-autosuggestions has 3 main behaviors: *accept* (accept the suggestion and update the buffer), *clear* (executes what's in the buffer) and *modify* (modify the buffer).  zsh-autosuggestions needs to wrap some magic around widgets and has some defaults for builtins widgets; by default, a widget is put into the *modify* "group".

`minimal-magic-enter` is a widget and is bound to the `^M` (`Enter` key usually) sequence, and if the buffer is not empty, it accepts the line. But zsh-autosuggestions puts it into the *modify* group while we want it to be into the *accept* group. 

The solution is a line away with configuring zsh-autosuggestions:

`ZSH_AUTOSUGGEST_CLEAR_WIDGETS+=minimal-magic-enter`

Make sure to update first

**Comment by samhh** [comment]
@subnixr No luck here, still seeing the same behaviour. I'm placing that line right here above L77: https://github.com/SamHH/dotfiles/blob/master/.zshrc#L77

Any ideas?

**Comment by subnixr** [comment]
I don't know about zplug, but you have to append to the array only after zsh-autosuggest is sourced (and minimal too?)

**Comment by subnixr** [comment]
Sorry, mistook the array name: `ZSH_AUTOSUGGEST_CLEAR_WIDGETS`. I've updated the previous comment as well

**Comment by samhh** [comment]
That's fixed it as soon as I placed it after `zplug load` (I guess that's when they're sourced), thanks a lot!

**Comment by samhh** [comment]
At some point in the last couple of months this bug came back for me despite having the above fix in my `.zshrc`. A regression I think? @subnixr 

Also I never use magic enter, so if there's a way to disable it and fix this as a temporary workaround I'd love to hear it (assuming that's still the issue)! :-)

**Comment by jmfederico** [comment]
@SamHH 

I changed the line to:
```bash
ZSH_AUTOSUGGEST_IGNORE_WIDGETS+=buffer-empty
```

And it worked for me. Do not know if anything will break.

@subnixr is that a reasonable solution?
